src/main/java/org/jitsi/jigasi/TranscriptionGateway.java

```diff
@@ -92,35 +140,51 @@ public void stop()
     }
 
     /**
-     * Tries to retrieve a transcriber assigned to a tenant
-     * if the property value is a json. Returns the value as
-     * is if no JSON is found.
+     * Tries to retrieve a custom defined transcriber by checking
+     * multiple sources. First it issues a GET request to the
+     * remoteTranscriptionConfigUrl property if defined. It expects
+     * a JSON response with a transcriberType property. If not found
+     * it tries to read the transcription.customService property. If
+     * that also fails it returns the default GoogleCloudTranscriptionService.
      *
      * @param tenant the tenant which is retrieved from the context
+     * @param roomJid the roomJid which is retrieved from the context (used only with JaaS)
      */
-    private String getCustomTranscriptionServiceClass(String tenant)
+    private String getCustomTranscriptionServiceClass(String tenant, String roomJid)
     {
         String transcriberClass = null;
-        String remoteTranscriptionConfigUrl
-                = JigasiBundleActivator.getConfigurationService()
-                .getString(
-                        REMOTE_TRANSCRIPTION_CONFIG_URL,
-                        null);
 
-        if (remoteTranscriptionConfigUrl != null && tenant != null)
+        if (remoteTranscriptionConfigUrl != null)
         {
-            String tsConfigUrl = remoteTranscriptionConfigUrl + "/" + tenant;
+            String tsConfigUrl;
+
+            // this is JaaS specific
+            if (remoteTranscriptionConfigUrl.contains("jitsi.net"))
+            {
+                tsConfigUrl = remoteTranscriptionConfigUrl + "?conferenceFullName="
+                        + URLEncoder.encode(roomJid, java.nio.charset.StandardCharsets.UTF_8);
+            }
+            else
+            {
+                String maybeTenant = tenant == null ? "" : tenant;
+                tsConfigUrl = remoteTranscriptionConfigUrl + "/" + maybeTenant;
+            }
+
             transcriberClass = getTranscriberFromRemote(tsConfigUrl);
+            logger.info("Transcriber class retrieved from remote " + remoteTranscriptionConfigUrl
+                    + ": " + transcriberClass);
         }
 
         if (transcriberClass == null)
         {
-            transcriberClass
-                    = JigasiBundleActivator.getConfigurationService()
+            String transcriberType = JigasiBundleActivator.getConfigurationService()
                     .getString(
                             CUSTOM_TRANSCRIPTION_SERVICE_PROP,
                             null);
+            transcriberClass = transcriberClasses.getOrDefault(transcriberType, null);
+            logger.info("Transcriber class retrieved from config: " + transcriberClass);
```

Comment:
So this will break existing configuration that use customService configured with a class, we need to make sure we support both or create a new one and deprecate the old one printing warnings and give time for people to adapt, for several months or an year, as we release not so often and that property was there for 5-6 years maybe.